javalib-ext-dummies/src/main/scala/java/util/Locale.scala

```diff
@@ -25,18 +32,49 @@ final class Locale(languageRaw: String, countryRaw: String)
 
   def getCountry(): String = country
 
+  def getVariant(): String = variant
+
+  def hasExtensions(): Boolean = !extensions.isEmpty()
+
+  def getExtension(key: Char): String = extensions.get(key) // nullable
+
+  // Not fully compliant, for debugging purposes only
   override def toString(): String = {
-    if (country == "") language
-    else language + "_" + country
+    var result = language
+    if (country != "" || variant != "")
+      result += "_" + country
+    if (variant != "")
+      result += "_" + variant
+
+    if (hasExtensions()) {
+      result += "#"
+      var first = true
+      var key = 'a'
+      while (key <= 'z') {
+        val value = getExtension(key)
+        if (value != null) {
+          if (first)
+            first = false
+          else
+            result += "-"
+          result += s"$key-$value"
+        }
+        key = (key + 1).toChar
+      }
```

Comment:
Do we really need to make this that low level? (We might even use the scalalib here, no?)

Reply:
Yes, good point. Fixed.